### ISSUE TITLE
visualization, train_classifier VS val_classifier

### ISSUE THREAD

**Issue by EvenGu** [hulianyuyy/STGAT#22]
Thank you for your great work.
For visualization, I wonder why the modification is made inside the train_classifer (line 51) instead of val_classifer? 
Please correct me if I misunderstood this. Modifying the train_classifier further trains and finetunes the model on a particular action class. It changes the attention weights. But given the pretrained model with frozen weights, is there a way to find out what joints are getting more attention in different actions? 
My second question relates to the attention weight to save. If Line 217 is the stgat_attention to save, what about [Line 227](https://github.com/hulianyuyy/STGAT/blob/f98db4df0e35243e858f6c6a47bdc99555e20156/model/st2ransformer_dsta.py#L227)? What is this diff doing?
I am looking forward to your reply. It will be much appreciated. Thanks!

**Comment by hulianyuyy** [comment]
Placing the modification in  train_classifer (line 51) is to visualize the classes in training samples. You may also place it in the val_classifier to show the attention weights. Maybe it's more proper to place it in the val_classifier without backward gradients. But my original goal is not to introduce gradients into this process.

Line 227 performs adaptively weighting for the attention weights. Here, i just want to show the original generated weights for the attention maps without the softmax operation.

**Comment by EvenGu** [comment]
I see. Thank you for the explanation!
